### PR TITLE
React Native 0.60 support

### PR DESCRIPTION
**Pull request by rlepinski**
Updates our module to work with React Native 0.60+ and newer. With .6, pods are included by default so we can simplify our setup instructions and sample. The main breaking change was in Android with the androidx migration.

They also changed how libraries autolink, so to take advantage of the new capabilities I recreated the sample app (now in example/*). We still no longer have to copy files around and we no longer have any weird hastemap errors.

I tested on both iOS and Android using the new example app and i tested it out in a standalone app from scratch.

Blog post on the new release:
https://facebook.github.io/react-native/blog/2019/07/03/version-60

CI will fail because I moved the ci script. Ill make sure it passes before I merge.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1 +1,3 @@
-sample/
+example
+.github
+scripts
```
Comment (marc-scig) [inline .npmignore]: is it possible this is causing the build failure you're currently seeing. At first glance it looks as though it can't find the ci script.

Comment (rlepinski) [reply .npmignore]: The build script use to be in the samples directory for whatever reason. I moved it to be a top level folder.

File: android/src/main/java/com/urbanairship/reactnative/Utils.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 
 import com.urbanairship.Logger;
 
+import static com.facebook.react.bridge.ReadableType.Null;
```
Comment (BrianBatchelder) [inline android/src/main/java/com/urbanairship/reactnative/Utils.java]: Is this cruft? It's unclear to me if it is used anywhere.

File: example/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+A basic sample application that integrates the Urban Airship React Native module.
+
+## Setup
+
+1) Install modules: Run `npm install` in repository root
+
+2) Create airship config files:
+  - iOS: example/ios/AirshipConfig.plist
+  - Android: example/android/app/src/main/assets/airshipconfig.properties
```
Comment (BrianBatchelder) [inline example/README.md]: Maybe add links to config docs?

File: example/android/app/BUCK [hunk 4]
```diff
@@ -35,12 +35,12 @@ android_library(
 
 android_build_config(
     name = "build_config",
-    package = "com.airshipsample",
+    package = "com.example",
```
Comment (BrianBatchelder) [inline example/android/app/BUCK]: com.airship.example?

File: example/android/app/build.gradle [hunk 5]
```diff
@@ -93,6 +96,15 @@ def enableSeparateBuildPerCPUArchitecture = false
  */
 def enableProguardInReleaseBuilds = false
 
+/**
+ * Use international variant JavaScriptCore
+ * International variant includes ICU i18n library and necessary data allowing to use
```
Comment (BrianBatchelder) [inline example/android/app/build.gradle]: ... includes _the_ ICU ...
... allowing ~to~ use _of methods_ ...

File: example/android/app/build.gradle [hunk 6]
```diff
@@ -93,6 +96,15 @@ def enableSeparateBuildPerCPUArchitecture = false
...
+ * e.g. Date.toLocaleString and String.localeCompare that give correct results
+ * when using with locales other than en-US.
```
Comment (BrianBatchelder) [inline example/android/app/build.gradle]: when ~using~ _used_ ...

File: example/android/app/build.gradle [hunk 7]
```diff
@@ -93,6 +96,15 @@ def enableSeparateBuildPerCPUArchitecture = false
...
+ * Note that this variant is about 6MiB larger per architecture than default.
```
Comment (BrianBatchelder) [inline example/android/app/build.gradle]: 6MB ?

File: example/android/app/src/main/java/com/urbanairship/sample/MainApplication.java [hunk 8]
```diff
@@ -22,15 +23,16 @@ public boolean getUseDeveloperSupport() {
 
     @Override
     protected List<ReactPackage> getPackages() {
-      return Arrays.<ReactPackage>asList(
-          new MainReactPackage(),
-            new ReactAirshipPackage()
-      );
+      @SuppressWarnings("UnnecessaryLocalVariable")
+      List<ReactPackage> packages = new PackageList(this).getPackages();
+      // Packages that cannot be autolinked yet can be added manually here, for example:
```
Comment (BrianBatchelder) [inline example/android/app/src/main/java/com/urbanairship/sample/MainApplication.java]: ... cannot _yet_ be autolinked ~yet~ ...

File: example/android/app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">example</string>
```
Comment (BrianBatchelder) [inline example/android/app/src/main/res/values/strings.xml]: Example?

File: package-lock.json [hunk 10]
```diff
@@ -1,627 +0,0 @@
-{
```
Comment (BrianBatchelder) [inline package-lock.json]: Why did you delete this file? Don't we usually keep it around for dependency history?

Comment (rlepinski) [reply package-lock.json]: We have a yarn.lock now. Going to try out yarn for this repo

File: example/android/app/build.gradle (hunk 5 above)
Comment (rlepinski) [reply example/android/app/build.gradle]: This is an autogenerated file
